### DISCUSSION TITLE
go-sdk is missing functionality to fetch SLI configs

### DISCUSSION THREAD

**Discussion by christian-kreuzberger-dtx** [keptn/keptn#8685] [General]
Several integrations for fetching SLIs use a function called `GetSLIConfiguration` which is part of go-utils: 
https://github.com/keptn/go-utils/blob/65503489d75aac53e74e089babc7a5258887d93c/pkg/lib/keptn/keptn_base.go#L83-L86

This function requires a `KeptnBase` instance to be present, which in return requires a cloud event to be present (among other things like KeptnOptions), e.g.:
```
// gotEvent processes an incoming CloudEvent
func gotEvent(event cloudevents.Event) error {
	keptnHandler, err := keptnv2.NewKeptn(&event, keptncommon.KeptnOpts{})

	customQueries, err := keptnHandler.GetSLIConfiguration(project, stage, service, utils.SliResourceURI)
	if err != nil {
		return nil, err
	}
...
```

By switching to the go-sdk approach for implementing handlers, we are losing this functionality, as

* It seems that it's not possible to get a "KeptnBase" object out of the sdk,
* It's not a good idea to use `NewKeptn`, as we're lacking the actual Cloud Event as well as keptn options (also, reconstructing the cloud event seems to be not the best idea... - but feel free to proof me wrong).

**Problem**

The lack of this implementation within go-sdk means that all services that are currently using `GetSLIConfiguration` need to re-implement this functionality based on the sdk. A first approach has already been done by dynatrace-service: 
https://github.com/keptn-contrib/dynatrace-service/blob/ddfadcff9a656974fe4e0d81c54c94602de3fce8/internal/keptn/config_client.go#L136-L172
(this also has `context.Context` as params, which is helpful)

This means that we will have multiple different implementations of this function, which will lead to increased maintenance costs for everyone involved in the future.

In addition, the `GetSLIConfiguration` will most likely no longer be-usable in go-utils, therefore we will have dead-code in go-utils.

**Evidence of Usage**

* *prometheus-service* 
* *sumologic-service* https://github.com/keptn-sandbox/sumologic-service/blob/54cf849cb74d89444d49e4e288dd1219716214df/eventhandlers.go#L157
* *datadog-service* https://github.com/keptn-sandbox/datadog-service/blob/6d7cf1883b8b194c9d4894708c1b7977f4f05495/eventhandlers.go#L84

**Proposal (please discuss)**

* Provide an easier way to use `GetSLIConfiguration` when using go-sdk

**Comment by warber** [comment]
I agree that this should not be implemented over and over again, but rather refactored/extracted into its own helper function that can be accessed independently of `KeptnBase` as a first step. `KeptnBase` in turn should be deprecated.

**Reply by arthurpitman** [reply]
The main motivation for the implementation in the dynatrace-service linked above was to add a `context.Context` to be used when accessing the underlying resources. Furthermore, once the dynatrace-service switched to using the controlplane connector, we no longer really needed to use `KeptnBase`.

**Reply by christian-kreuzberger-dtx** [reply]
@arthurpitman @warber  I'd suggest that the new implementation also has a  `context.Context` parameter, and uses the new SDK functions for fetching files. 

Edit: Would it be possible to move https://github.com/keptn-contrib/dynatrace-service/blob/ddfadcff9a656974fe4e0d81c54c94602de3fce8/internal/keptn/config_client.go#L128-L172 into go-sdk?
